### PR TITLE
chores/application-configuration-for-test-setup 

### PR DESCRIPTION
**Pull request by laujonat**
What was done: 
```
Addressed  jest/enzyme configuration issues with Babel transpiler.
Basic test using for `Loading` with snapshots
Cleaned up some code around the express routes 
Simple snapshot enzyme test for initial rendered UI
Added testing for reducer actions in map_reducer 
Add npm package redux-saga-test-plan
Moved jest.config.js into package.json just because
Moved routes directory structure. We don't have a "server" 
Separated loading indicators for the separate entities (map/lyft)
```

Too much was done regarding configuration/ test env setup that probably should have been done in another PR.  Don't want to continue adding changes before PR review. We should probably just create a new branch for `draw-boundaries` after merging this back into development to keep the PR small. 

@xvzero

### CODE REVIEW

**Comment by laujonat** [comment]
@mingzhaogu @xvzero This initially started out to refactor the async calls. But along the way there were a lot of other things that had to be addressed. I'm just suggesting we merge this into development and we can create another `chores/draw-boundaries-saga` with its original intention.

File: client/components/UserInterface/UserInterface.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Map from "../Map"
+import { shallow } from "enzyme"
+
+describe("UserInterface", () => {
+  it("renders the NavPane with no props", () => {
+    const NavPaneComponent = shallow(<NavPane />)
+    expect(NavPaneComponent).toMatchSnapshot()
+  })
+
+  it("renders the NavPane with no props", () => {
```
Comment (xvzero) [inline client/components/UserInterface/UserInterface.test.js]: This should be moved to `Map.test.js`

**Comment by laujonat** [comment]
@xvzero thoughts on folder structure change?

File: client/reducers/map_reducer.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    })
+  })
+
+  it("should handle RECEIVE_MARKER_LOCATION_ERROR", () => {
+    const mockData = { data: "junk" }
+    const action = {
+      type: RECEIVE_MARKER_LOCATION_ERROR,
+      data: mockData.data
+    }
+    expect(mapReducer(initialState, action)).toEqual({
```
Comment (xvzero) [inline client/reducers/map_reducer.test.js]: Shouldn't there be some `errors` attribute in the state?

File: client/components/UserInterface/UserInterface.test.js (hunk 1 above)
Comment (laujonat) [reply client/components/UserInterface/UserInterface.test.js]: addressed

File: client/reducers/map_reducer.test.js (hunk 2 above)
Comment (laujonat) [reply client/reducers/map_reducer.test.js]: addressed

**Pull request merged**